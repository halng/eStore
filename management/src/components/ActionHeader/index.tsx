'use client'

import { Box, Button } from '@mui/material'

import * as React from 'react'
import TextField from '@mui/material/TextField'
import Autocomplete from '@mui/material/Autocomplete'
import { ProductGroupTableData } from '@types'
import AddIcon from '@mui/icons-material/Add'
import CustomBreadcrumbs from '../Breadcrumbs'

interface SearchBoxProps {
    initializeData: ProductGroupTableData[]
}

const SearchBox = ({ initializeData }: SearchBoxProps) => {
    return (
        <Autocomplete
            sx={{ width: '40%', height: '3rem' }}
            freeSolo
            id='free-solo-2-demo'
            disableClearable
            options={(initializeData || []).map((option) => option.name)}
            renderInput={(params) => (
                <TextField
                    {...params}
                    label='Search'
                    InputProps={{
                        ...params.InputProps,
                        type: 'search',
                    }}
                />
            )}
        />
    )
}

interface ActionHeaderProps {
    tableData: ProductGroupTableData[]
}

const ActionHeader = ({ tableData }: ActionHeaderProps) => {
    return (
        <Box sx={{ display: 'flex', alignItems: 'left', mb: 2, flexDirection: 'column', mt: 3 }}>
            <CustomBreadcrumbs />
            <Box
                sx={{
                    display: 'flex',
                    width: '100%',
                    justifyContent: 'space-between',
                    paddingRight: '1rem',
                    pt: '2rem',
                }}
            >
                {<SearchBox initializeData={tableData} />}
                <Button variant='contained' color='success' startIcon={<AddIcon />}>
                    Add
                </Button>
            </Box>
        </Box>
    )
}

export default ActionHeader
