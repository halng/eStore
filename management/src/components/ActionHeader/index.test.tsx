import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import ActionHeader from '.'

jest.mock('next/navigation', () => ({
    usePathname: jest.fn(() => '/partner/test'),
}))

describe('Test Action Header Component', () => {
    it('snapshot rendering', () => {
        const { asFragment } = render(<ActionHeader tableData={[]} setOpenAction={null} />)
        expect(asFragment()).toMatchSnapshot()
    })
    it('renders a heading', () => {
        render(<ActionHeader tableData={[]} setOpenAction={null} />)

        // check render breadcrumbs
        const separatorCount = screen.getAllByText('/')
        expect(separatorCount).toHaveLength(2)

        const homeIcon = screen.getByTestId('HomeIcon')
        expect(homeIcon).toBeInTheDocument()

        const pathElement = screen.getByText('partner', { exact: false })
        expect(pathElement).toBeInTheDocument()

        // check render search box
        const searchBox = screen.getByRole('combobox')
        expect(searchBox).toBeInTheDocument()

        // check render create button
        const createButton = screen.getByRole('button', { name: 'Add' })
        expect(createButton).toBeInTheDocument()
    })
})
