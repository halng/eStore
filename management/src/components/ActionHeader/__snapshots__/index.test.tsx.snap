// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Action Header Component snapshot rendering 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-qlplcg"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-bvyohc-MuiTypography-root-MuiLink-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-bvyohc-MuiTypography-root-MuiLink-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
            >
              partner
            </p>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-bvyohc-MuiTypography-root-MuiLink-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1dzd4o2-MuiTypography-root"
            >
              test
            </p>
          </a>
        </li>
      </ol>
    </nav>
    <div
      class="MuiBox-root css-4tea8"
    >
      <div
        class="MuiAutocomplete-root css-4lnbko-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="free-solo-2-demo"
            id="free-solo-2-demo-label"
          >
            Search
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiAutocomplete-inputRoot css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiAutocomplete-input MuiAutocomplete-inputFocused css-1g24dm6-MuiInputBase-input-MuiOutlinedInput-input"
              id="free-solo-2-demo"
              role="combobox"
              spellcheck="false"
              type="search"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Search
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess css-ke5b6m-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
            />
          </svg>
        </span>
        Add
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
